src/linear_algebra/bilinear_form.lean

```diff
@@ -1016,3 +1078,250 @@ begin
+  have : ∀ u v, 2 * B u v = 0,
+  { intros u v,
+    rw [show 2 * B u v = B u u + B v u + B u v + B v v,
+          by rw [is_ortho_def.1 (h u), is_ortho_def.1 (h v), hB₂ v u, two_mul, zero_add]; ring,
+        show B u u + B v u + B u v + B v v = B (u + v) (u + v),
+          by simp [← add_assoc], is_ortho_def.1 (h _)] },
+  have hcon : ∀ u v, B u v = 0,
+  { intros u v,
+    rw [show 0 = htwo.inv_of * (2 * B u v), by rw [this, mul_zero]], simp [← mul_assoc] },
+  exact let ⟨v, hv⟩ := exists_ne (0 : M) in hv $ hB₁ v (hcon v),
+end
+
+-- ↓ This lemma only applies in fields as we require `a * b = 0 → a = 0 ∨ b = 0`
+lemma span_singleton_inf_orthogonal_eq_bot
+  {B : bilin_form K V} {x : V} (hx : ¬ B.is_ortho x x) :
+  (K ∙ x) ⊓ B.orthogonal (K ∙ x) = ⊥ :=
+begin
+  rw ← finset.coe_singleton,
+  refine eq_bot_iff.2 (λ y h, _),
+  rcases mem_span_finset.1 h.1 with ⟨μ, rfl⟩,
+  have := h.2 x _,
+  { rw finset.sum_singleton at this ⊢,
+    suffices hμzero : μ x = 0,
+    { rw [hμzero, zero_smul, submodule.mem_bot] },
+    change B (μ x • x) x = 0 at this, rw [smul_left] at this,
+    exact or.elim (zero_eq_mul.mp this.symm) id (λ hfalse, false.elim $ hx hfalse) },
+  { rw submodule.mem_span; exact λ _ hp, hp $ finset.mem_singleton_self _ }
+end
+
+-- ↓ This lemma only applies in fields since we use the inverse
+lemma span_singleton_sup_orthogonal_eq_top {B : bilin_form K V}
+  (hB : sym_bilin_form.is_sym B) {x : V} (hx : ¬ B.is_ortho x x) :
+  (K ∙ x) ⊔ B.orthogonal (K ∙ x) = ⊤ :=
+begin
+  refine eq_top_iff.2 (λ y _, _),
+  rw submodule.mem_sup,
+  refine ⟨(B x y * (B x x)⁻¹) • x, _, y - (B x y * (B x x)⁻¹) • x, _, _⟩,
+  { exact submodule.mem_span_singleton.2 ⟨(B x y * (B x x)⁻¹), rfl⟩ },
+  { intros z hz,
+    rcases submodule.mem_span_singleton.1 hz with ⟨μ, rfl⟩,
+    simp [is_ortho, mul_assoc, inv_mul_cancel hx, hB x] },
+  { simp }
+end
+
+lemma span_singleton_is_compl_orthogonal {B : bilin_form K V}
+  (hB : sym_bilin_form.is_sym B) {x : V} (hx : ¬ B.is_ortho x x) :
+  is_compl (K ∙ x) (B.orthogonal (K ∙ x)) :=
+⟨le_bot_iff.2 $ span_singleton_inf_orthogonal_eq_bot hx,
+  top_le_iff.2 $ span_singleton_sup_orthogonal_eq_top hB hx⟩
+
+/-- Given a bilinear form `B` and some `x` such that `B x x ≠ 0`, the span of the singleton of `x`
+  is complement to its orthogonal complement. -/
+lemma is_compl_span_singleton_orthogonal {B : bilin_form K V}
+  (hB : sym_bilin_form.is_sym B) {x : V} (hx : ¬ B.is_ortho x x) :
+  is_compl (K ∙ x) (B.orthogonal $ K ∙ x) :=
+{ inf_le_bot := eq_bot_iff.1 $ span_singleton_inf_orthogonal_eq_bot hx,
+  top_le_sup := eq_top_iff.1 $ span_singleton_sup_orthogonal_eq_top hB hx }
+
+/-- The natural isomorphism between a singleton and the quotient by its orthogonal complement. -/
+noncomputable def quotient_equiv_of_ortho_singleton
+  {B : bilin_form K V} (hB : sym_bilin_form.is_sym B) {x : V} (hx : ¬ B.is_ortho x x) :=
+submodule.quotient_equiv_of_is_compl _ _ (is_compl_span_singleton_orthogonal hB hx)
```

Comment:
Likewise I'd delete this lemma now that you have a proof of the `dim + 1` formula which bypasses it.